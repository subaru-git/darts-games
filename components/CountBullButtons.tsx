import React, { FC } from "react";
import BullButton from "./BullButton";

type CountBullButtonsProps = {
  onCount: (count: point) => void;
  disabled: boolean;
};

const CountBullButtons: FC<CountBullButtonsProps> = ({ onCount, disabled }) => {
  return (
    <>
      <BullButton onScore={(n) => onCount(n)} disable={disabled} />
    </>
  );
};

export default CountBullButtons;
