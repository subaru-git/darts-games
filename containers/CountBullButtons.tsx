import React, { FC } from "react";
import BullButton from "../components/BullButton";

type CountBullButtonsProps = {
  onCount: (count: string) => void;
  disabled: boolean;
};

const CountBullButtons: FC<CountBullButtonsProps> = ({ onCount, disabled }) => {
  return (
    <>
      <BullButton onScore={(n) => onCount(n)} disable={disabled} />
    </>
  );
};

export default CountBullButtons;
