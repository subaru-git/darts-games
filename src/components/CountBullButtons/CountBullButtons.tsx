import React, { FC } from 'react'
import { css } from '@emotion/react'

type CountBullButtonsProps = {
  onCount: (n: point) => void
  disabled?: boolean
}

const CountBullButtons: FC<CountBullButtonsProps> = ({ onCount, disabled = false }) => {
  return (
    <div css={styles}>
      <div css={[disableStyle, disabled ? null : css({ display: 'none' })]} />
      <button
        css={outerBullStyles}
        onClick={() => onCount('S-BULL')}
        disabled={disabled}
        aria-label='outer bull'
      />
      <button
        css={innerBullStyles}
        onClick={() => onCount('D-BULL')}
        disabled={disabled}
        aria-label='inner bull'
      />
      <button
        css={nonBullStyles}
        onClick={() => onCount('0')}
        disabled={disabled}
        aria-label='non bull'
      />
    </div>
  )
}

const styles = css({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  height: '40vw',
  Width: '40vw',
  maxWidth: '40vh',
  maxHeight: '40vh',
})
const innerBullStyles = css({
  position: 'absolute',
  borderRadius: '100%',
  height: '15vw',
  width: '15vw',
  maxWidth: '15vh',
  maxHeight: '15vh',
  color: 'white',
  backgroundImage: 'radial-gradient(circle, #111111 0% 40%, #666666 90% 100%)',
  zIndex: 3,
  WebkitAppearance: 'none',
  '&:hover': {
    backgroundImage: 'radial-gradient(circle, #111111 0% 40%, #444444 90% 100%)',
  },
  '&:active::after': {
    content: '""',
    position: 'absolute',
    borderRadius: '100%',
    height: '15vw',
    width: '15vw',
    maxWidth: '15vh',
    maxHeight: '15vh',
    marginTop: '-50%',
    marginLeft: '-50%',
    verticalAlign: 'text-top',
    background: '#777777',
    opacity: 0,
    animation: 'ripple-in 0.3s ease-out',
  },
  '@keyframes ripple-in': {
    '0%': {
      opacity: 0.8,
      transform: 'scale(1.3)',
    },
    '100%': {
      opacity: 0,
      transform: 'scale(0.0)',
    },
  },
})
const outerBullStyles = css({
  position: 'absolute',
  borderRadius: '100%',
  height: '30vw',
  width: '30vw',
  maxWidth: '30vh',
  maxHeight: '30vh',
  verticalAlign: 'text-top',
  backgroundImage: 'radial-gradient(circle, #ff2222 0% 60%, #ff6666 90% 100%)',
  WebkitAppearance: 'none',
  zIndex: 2,
  '&:hover': {
    backgroundImage: 'radial-gradient(circle, #ff0000 0% 60%, #ff4444 90% 100%)',
  },
  '&:active::after': {
    content: '""',
    position: 'absolute',
    borderRadius: '100%',
    height: '30vw',
    width: '30vw',
    maxWidth: '30vh',
    maxHeight: '30vh',
    marginTop: '-50%',
    marginLeft: '-50%',
    verticalAlign: 'text-top',
    background: '#dd7777',
    opacity: 0,
    animation: 'ripple-out 0.3s ease-out',
  },
  '@keyframes ripple-out': {
    '0%': {
      opacity: 0.8,
      transform: 'scale(0.6)',
    },
    '100%': {
      opacity: 0,
      transform: 'scale(1.3)',
    },
  },
})
const nonBullStyles = css({
  height: '40vw',
  width: '40vw',
  WebkitAppearance: 'none',
  backgroundImage: 'radial-gradient(circle, #777777 0% 70%, #666666 90% 100%)',
  zIndex: 1,
  maxWidth: '40vh',
  maxHeight: '40vh',
  '&:hover': {
    backgroundImage: 'radial-gradient(circle, #666666 0% 70%, #555555 90% 100%)',
  },
})
const disableStyle = css({
  backgroundColor: '#222222',
  width: '40vw',
  height: '40vw',
  position: 'absolute',
  zIndex: 4,
  opacity: '50%',
  maxWidth: '40vh',
  maxHeight: '40vh',
})

export default CountBullButtons
