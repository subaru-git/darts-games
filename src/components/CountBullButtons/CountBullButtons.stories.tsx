import type { Meta, StoryObj } from '@storybook/react';
import CountBullButtons from './CountBullButtons';

const meta: Meta<typeof CountBullButtons> = {
  component: CountBullButtons,
  title: 'Components/CountBullButtons',
};

export default meta;
type Story = StoryObj<typeof CountBullButtons>;
export const Default: Story = { args: { disabled: false } };
