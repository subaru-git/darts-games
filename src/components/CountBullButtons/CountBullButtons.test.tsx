import { fireEvent, render } from '@testing-library/react';
import CountBullButtons from './CountBullButtons';

test('should rendering and clickable input', () => {
  const mockOnScore = jest.fn();
  const { container } = render(<CountBullButtons onCount={mockOnScore} />);
  fireEvent.click(container.querySelector('button')!);
  expect(mockOnScore).toBeCalledTimes(1);
  expect(container).toMatchSnapshot();
});

test('should rendering and disable input', () => {
  const mockOnScore = jest.fn();
  const { container } = render(<CountBullButtons onCount={mockOnScore} disabled />);
  fireEvent.click(container.querySelector('button')!);
  expect(mockOnScore).toBeCalledTimes(0);
  expect(container).toMatchSnapshot();
});
