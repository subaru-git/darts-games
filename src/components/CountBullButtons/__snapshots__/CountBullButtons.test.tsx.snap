// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering and clickable input 1`] = `
<div>
  <div
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    <div
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).,You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
    <button
      aria-label="outer bull"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
    <button
      aria-label="inner bull"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
    <button
      aria-label="non bull"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
  </div>
</div>
`;

exports[`should rendering and disable input 1`] = `
<div>
  <div
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    <div
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop).,"
    />
    <button
      aria-label="outer bull"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      disabled=""
    />
    <button
      aria-label="inner bull"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      disabled=""
    />
    <button
      aria-label="non bull"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      disabled=""
    />
  </div>
</div>
`;
